File: kotlin_study/src/main/java/ex08/Exam08.java
package ex08;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Optional;

public class Exam08 {

    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy MM dd");
    private final String stringRegisterAt;

    public Exam08(Store store){
        this.stringRegisterAt = toLocalDateTimeString(store.getRegisterAt());
    }

    public String toLocalDateTimeString(LocalDateTime localDateTime){
        return Optional.ofNullable(localDateTime)
            .orElseGet(LocalDateTime::now)
            .format(FORMATTER);
    }

    public String getFormattedRegisterAt() {
        return stringRegisterAt;
    }

    public static void main(String[] args) {
        var registerDto = new Store();
        Exam08 exam08 = new Exam08(registerDto);
        System.out.println(exam08.getFormattedRegisterAt());
    }

}

class Store {

    private LocalDateTime registerAt;

    public LocalDateTime getRegisterAt() {
        return registerAt;
    }

    public void setRegisterAt(LocalDateTime registerAt) {
        this.registerAt = registerAt;
    }

}

File: kotlin_study/src/main/kotlin/ex08/Exam08.kt
package org.example.ex08

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

fun main(args: Array<String>) {
    Exam08(Store())
    println(DateTimeUtil().localDateTimeToString())
}

data class Store(
    var registerAt: LocalDateTime ?= null
)

class Exam08(store: Store) {

    init {
        val strLocalDateTime = toLocalDateTimeString(store.registerAt)
        println(strLocalDateTime)

        println(toLocalDateTimeString())
    }

    fun toLocalDateTimeString(
        date: LocalDateTime?=LocalDateTime.of(2024, 1, 1, 1, 1)
    ): String {
        val temp = date?:LocalDateTime.now()
        return temp.format(DateTimeFormatter.ofPattern("yyyy MM dd"))
    }

}

class DateTimeUtil {

    private val KST_FORMAT = "yyyy년 MM월 dd일 HH시 mm분 ss초"

    fun localDateTimeToString(
        localDateTime: LocalDateTime?=LocalDateTime.now(),
        pattern: String?=KST_FORMAT
    ): String {
        val temp = localDateTime?:LocalDateTime.now()
        return temp.format(
            pattern?.let { DateTimeFormatter.ofPattern(it) }
        )
    }

}